Field renaming and docs tidy

robbrockbank opened the pull request.

gunjan5 commented on calicoctl/commands/node/diags.go:
```diff
@@ -84,7 +92,7 @@ func runDiags(logDir string) {
 
 	// Create a temp directory in /tmp
-	tmpDir, err := ioutil.TempDir("/tmp", "tmp")
+	tmpDir, err := ioutil.TempDir("", "calico")
```
good catch!

gunjan5 commented on Makefile:
```diff
@@ -191,7 +191,8 @@ test-containerized: run-etcd $(BUILD_CONTAINER_MARKER)
 	-e PLUGIN=calico \
 	-v ${PWD}:/go/src/github.com/projectcalico/calico-containers:rw \
-	$(BUILD_CONTAINER_NAME) make ut
+	$(BUILD_CONTAINER_NAME) make ut; \
```
This line should be `$(BUILD_CONTAINER_NAME) bash -c 'make ut; \` 
That should fix the CircleCI build

robbrockbank commented:
The `--version` issue should really be another PR, but I've fixed up here since it was easy.  I have fixed up to only allow `--version` from the  root calicoctl command, not the sub commands.

It returns a version summary string, much like Dockers.

So, `calicoctl --version` will work, but `calicoctl apply --version` will print a usage statement.
